mayastor/src/rebuild.rs

```diff
@@ -331,13 +340,47 @@ impl RebuildActions for RebuildTask {
         });
         complete_receiver
     }
-    fn stop(&mut self) {
-        self.change_state(RebuildState::Stopped);
+
+    fn stop(&mut self) -> Result<(), RebuildError> {
+        match self.state {
+            RebuildState::Paused => {
+                self.change_state(RebuildState::Stopped);
+                // The rebuild is paused so call complete here
+                // because the run function is inactive
+                self.send_complete();
+            }
+            _ => self.change_state(RebuildState::Stopped),
+        }
+
+        Ok(())
     }
-    fn pause(&mut self) {
-        todo!("pause the rebuild task");
+
+    fn pause(&mut self) -> Result<(), RebuildError> {
+        match self.state {
+            RebuildState::Running => {
+                self.change_state(RebuildState::Paused);
+                Ok(())
+            }
+            _ => Err(RebuildError::OpError {
```

Comment:
The error seems to be repeated a lot, could we have a "builder" that returns the error?, eg:
fn operation_error(next) { 
return 
Err(RebuildError::OpError {
                operation: next.to_string(),
                state: self.state.to_string(),
            })
}
_ => Self::operation_error(RebuildState::Paused);

Reply:
After trying various options out, I think it best to leave this as it is. Elsewhere in the code where we use the same errors we set up the error structure at the point of returning, so this way is consistent with the rest of the code.